src/runtime/composables/search.ts

```diff
@@ -0,0 +1,100 @@
+import MiniSearch, { type Options as MiniSearchOptions } from 'minisearch'
+import { useRuntimeConfig, useFetch } from '#imports'
+
+export const useIndexedSearch = async <DataItem>(search: MaybeRefOrGetter<string>) => {
+  const runtimeConfig = useRuntimeConfig()
+  const integrity = runtimeConfig.public.content.integrity
+  const baseAPI = runtimeConfig.public.content.api.baseURL
+  const { options } = runtimeConfig.public.content.search
+
+  const { data } = await useFetch<DataItem[]>(`${baseAPI}/indexed-search${integrity ? '.' + integrity : ''}.txt`)
+
+  if (!data.value) { return [] }
+
+  const { results } = useIndexedMiniSearch(search, data as unknown as string, options)
+
+  return results
+}
+
+export const defineMiniSearchOptions = <DataItem>(options: MiniSearchOptions<DataItem>) => {
+  return ref(options)
+}
+
+export const useSearch = async <DataItem>(search: MaybeRefOrGetter<string>, options: MiniSearchOptions<DataItem>) => {
```

Comment:
Would be great if we merge `useIndexedSearch` and `useSearch` into a single composable. Having two composables make no sense because the API works based on `indexedSearch` boolean

Reply:
I use runtimeConfig!